### ISSUE TITLE
Showing  the found streams in FEN takes a verrrrrrrrrry long time to show

### ISSUE THREAD

**Issue by peno64** [Tikipeter/repository.tikipeter#180]
For example when I search for the tv shows start trek picard with a real debrid account, it find *alot of streams. About 500. But when all providers are done it looks like fen is then doing nothing and returns to the episode overview. But if you wait long enough and I mean really long being more than 5 or 10 minutes! The list of found links is shown. I have this problem only in FEN do it is not my device. The device is a raspberry pi 4 with 4 gb of memory. The overall performance of it is very good. It is only here that there is a performance issue.
I am quite technical so if you want me to try something with a new file or some logging, no problem for me.

**Comment by Tikipeter** [comment]
Hi peno64.

A few questions/suggestions:
1. Do you have Fen set to check for cached torrents? If so, does this happen immediately at the end of scraping (you will see the progress dialog)?
2. Have you tried restarting your Pi to see if it fixes this issue (I know you shouldn't have to do this, but it's good to see for troubleshooting)?
3. Do you have Settings->General->Navigation->"Use Threading for Menu Items" enabled? If so, turn it OFF and see if that fixes your issue.
4. Make sure you are on the latest version of Fen (v.1.6.11)

Some things to try:
* Settings->Results->Display - set "Results Display" to VERY SIMPLE DIRECTORY and see if that fixes the issue.
* If not, try setting it to DIALOG and see if that fixes the issue.

If none of those suggestions work, and probably even if they do work, I will most likely need a log file.

**Comment by peno64** [comment]
- Check Debrid Services for Cached Torrents is off
- I already restarted the pi and also with another sd card but also the same problem. I also tried to clear all cache and to remove fen and its dependecies and reinstall it but that also didn't help
- Use Threading for Menu Items was already off
- I do have the latest version of Fen: 1.6.11

Thank you

**Comment by peno64** [comment]
A small update. Even with a tv show with only a few streams (6) also takes minutes before the streams are shown. Strange, isn't it?

**Comment by Tinuz97** [comment]
Strange, yes.
Asked a friend with a raspberry pi 4, same settings as I use with Fen (cached only enabled, debrid only, remove duplicates also enabled), and fast for him. 

Do you have Realdebrid, if yes, maybey you can try to test that also?

**Comment by peno64** [comment]
I do have real debrid. I will try on another sd card with a new installation of kodi and only fen if I have also the issue. If not it must be because some other addon that is installed. Maybe this weekend I have some time to test this

**Comment by Tinuz97** [comment]
I use these at results settings (I have a nvidia shield), but a buddy with a raspberry pi 4 is using the same settings (he does not use the torrent file size filter, so you could ignore it, but less result=faster, I don't use movies below 3gb,so it does filter them out, is less results=faster) 

I am realy interested if it does speed things up for you, like my buddy's raspberry pi 4. 


![Screenshot_20200402-195558](https://user-images.githubusercontent.com/47715606/78282150-1a5c1d80-751c-11ea-8959-a5d8da801409.jpg)
![Screenshot_20200402-195611](https://user-images.githubusercontent.com/47715606/78282158-1defa480-751c-11ea-8dbb-374e68395676.jpg)
![Screenshot_20200402-195623](https://user-images.githubusercontent.com/47715606/78282170-221bc200-751c-11ea-9f52-4401594c022f.jpg)

**Comment by Tinuz97** [comment]
@Tikipeter, also openscrapers needs realy a cfscrape update, CloudFare does use Hcapthca since a few days, so struggling with some scrapers, maybey you could inform the devs of openscrapers for this?

I already replaced the files as a hotfix, but for other users, VeNoMouS has updated it a few days ago. 


detect reCaptcha and hCaptcha

https://github.com/VeNoMouS/cloudscraper/tree/dev

**Comment by peno64** [comment]
I created a brand new kodi install on another sd card and only installed fen and now it works just fine. So it must be because of some other addons...
I am now copying a complete image of the sd card I normally use to a new sd card and will remove addons one by one until I have found which addon causes this problem.

**Comment by peno64** [comment]
That is why I love the pi so much. Just put another sd card in it and you can experiment as much as you want. Putting the original card back in and you are running within a minute again.

**Comment by Tinuz97** [comment]
I have a local repo and wizard, So I can experiment also without issues 😄

**Comment by peno64** [comment]
@Tinuz97 please elaborate. Interested....

**Comment by Tinuz97** [comment]
Repo (including manual)

https://github.com/drinfernoo/repository.example

Wizard (including manual) 

https://github.com/drinfernoo/plugin.program.openwizard

And just run a simple webserver on a local computer (online also possible of course, but i have it local for my own, not public (Wan).

**Comment by Tikipeter** [comment]
If it's a conflicting add-on causing issues, I'd love to know what that add-on is so I can install it and see what's going on.  There have been troubles in the past similar to this, but they have been resolved over time, so I'm at a loss as to the problem. Hopefully when you reload all your previous add-ons, it will 'just work', and can be put up to a Kodi quirk.

**Comment by peno64** [comment]
Working on it. I am uninstalling addons until it is ok again. I already uninstalled them all except for fen and it was ok again so yes it must be a conflicting addon but it takes time to discover which one...

**Comment by peno64** [comment]
ok I came to a point I unstalled every addon I could, even via system, add-ons, manage dependencies except for the fen dependencies and that did not help at all. Of course I first tried to clear all cashes via the settings in Fen but that all didn't help anything.
Finaly I did this:
- uninstall Fen with remove all settings
- kodi system, add-ons, manage dependencies:
  - uninstall openscrapers module and all related data
  - uninstall Tiki Artwork
  - uninstall Tiki Meta Module and all related data
- reboot
- reinstall Fen from the repo I still had on kodi
- reauthorize Real Debrid & trakt

And now it is ok again.
